### PR TITLE
Create "Integration Patterns" section in docs

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation

> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

This sets us up to write the series of guides for Game Server Allocation advanced filtering (#1239), as well as adding
extra guides for integrations such as:

* Websocket servers
* Reusing GameServers
* Canary Testing

And I'm sure more will show up!

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

N/A

### CODE REVIEW

**Comment by markmandel** [comment]
Tracking Failure I need to fix:

```
Html Test: Attempt 1
htmltest started at 12:58:36 on /tmp/website
========================================================================
site/docs/integration-patterns/matchmaker-registration/index.html
  target does not exist --- site/docs/integration-patterns/matchmaker-registration/index.html --> ../../../../diagrams/gameserver-reserved.puml.png
========================================================================
✘✘✘ failed in 307.854464ms
```

File: site/content/en/docs/Integration Patterns/matchmaker-registration.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+In this scenario, the `GameServer` process will need to self Allocate when informed by the matchmaker that players 
+are being sent to them.
+
+![Reserved Lifecycle Sequence Diagram](../../../diagrams/gameserver-reserved.puml.png)
+
+{{< alert title="Warning" color="warning">}}
+This does relinquish control over how `GameServers` are packed across the cluster to the external matchmaker. It is likely
+it will not do as good a job at packing and scaling as Agones.
+{{< /alert >}}
+gameserver-reserved.puml.png
```
Comment (roberthbailey) [inline site/content/en/docs/Integration Patterns/matchmaker-registration.md:20]: This line seems extraneous (and is rendered in the page).

**Review by roberthbailey — COMMENTED** [review]
Just one nit, you can see the extra text on https://88341a4-dot-preview-dot-agones-images.appspot.com/site/docs/integration-patterns/matchmaker-registration/

File: site/content/en/docs/Integration Patterns/matchmaker-registration.md (hunk 1 above)
Comment (markmandel) [reply site/content/en/docs/Integration Patterns/matchmaker-registration.md]: Looks like I left a bad copy/paste! Fixed 👍🏻

**Review by roberthbailey — APPROVED** [review]

**Pull request merged**